pkg/util/cloudfoundry/cccache.go

```diff
@@ -0,0 +1,163 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build clusterchecks
+
+package cloudfoundry
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/cloudfoundry-community/go-cfclient"
+)
+
+// CCCacheI is an interface for a structure that caches and automatically refreshes data from Cloud Foundry API
+// it's useful mostly to be able to mock CCCache during unit tests
+type CCCacheI interface {
+	LastUpdated() time.Time
+	GetPollAttempts() int
+	GetPollSuccesses() int
+	GetApp(string) (*cfclient.V3App, error)
+}
+
+// CCCache is a simple structure that caches and automatically refreshes data from Cloud Foundry API
+type CCCache struct {
+	sync.RWMutex
+	cancelContext context.Context
+	configured    bool
+	ccAPIClient   CCClientI
+	pollInterval  time.Duration
+	pollAttempts  int
```

Comment:
We should use atomic operation instead of lock for `pollAttempts` and `pollSuccesses`

Reply:
What's the benefit ?

Reply:
Locking is slower that atomic operation. Plus since we share the same lock we have contention when accessing 2 different variables.

Reply:
done